pkg/plugin/v3/scaffolds/internal/templates/config/crd/kustomizeconfig.go

```diff
@@ -35,23 +40,52 @@ func (f *KustomizeConfig) SetTemplateDefaults() error {
 		f.Path = filepath.Join("config", "crd", "kustomizeconfig.yaml")
 	}
 
-	f.TemplateBody = kustomizeConfigTemplate
+	switch f.CRDVersion {
+	case v1beta1:
+		f.TemplateBody = kustomizeConfigTemplatev1beta1
+	default:
+		f.TemplateBody = kustomizeConfigTemplatev1
+	}
```

Comment:
You are templating at the `SetTemplateDefault` method. Let's try to keep the templating logic at the template.
```suggestion
	if f.CRDVersion == "" {
		f.CRDVersion = v1
	}
	f.TemplateBody = kustomizeConfigTemplate
```

This, and the following two changes, also apply to:
- `config/crd/patches/enablecainjection_patch.go`
- `config/crd/patches/enablewebhook_patch.go`: with a little extra modification
- `config/kdefault/enablecainection_patch.go`: by the way, can you fix the typo in the name?

Reply:
See https://github.com/kubernetes-sigs/kubebuilder/pull/1644#discussion_r524683792